File: Python/July_problems/problem1190_11_july_2024.py
def reverseParentheses(s: str) -> str:
    stack = []
    current_string = ""
    for char in s:
        if char == '(':
            stack.append(current_string)
            current_string = ""
        elif char == ')':
            current_string = current_string[::-1]
            current_string = stack.pop() + current_string
        else:
            current_string += char
    return current_string


def problem1190():
    pass


if __name__ == "__main__":
    problem1190()

File: CPP/Year-2024/July_Problems/problem1190_11_july_2024.cpp
#include <iostream>
#include <stack>
#include <algorithm>

using namespace std;

string reverseParentheses(string s)
{
    stack<string> S;
    string ans = "";
    for (char ch : s)
    {
        if (ch == '(')
        {
            S.push(ans);
            ans = "";
        }
        else if (ch == ')')
        {
            reverse(ans.begin(), ans.end());
            ans = S.top() + ans;
            S.pop();
        }
        else
        {
            ans += ch;
        }
    }
    return ans;
}

int main()
{
    return 0;
}